src/main/java/org/keycloak/webbuilder/builders/AppBuilder.java

```diff
@@ -3,49 +3,94 @@
+        // Get package contents as tarball stream.
+        Package packageInfo = Registry.getPackage(name);
+        Version latestVersion = packageInfo.getVersionByTag(version);
         ArchiveInputStream<TarArchiveEntry> tarball = latestVersion.getDist().getTarballStream();
 
-        for (ArchiveEntry entry = tarball.getNextEntry(); entry != null; entry = tarball.getNextEntry()) {
-            // Loop trough until we find a file that matches the package entrypoint.
-            if (!entry.getName().equals(sourcePath)) {
+        // Copy package contents to installation path.
+        ArchiveEntry entry;
+        while ((entry = tarball.getNextEntry()) != null) {
+            // Skip any files not part of the package contents.
+            String packagePrefix = "package";
+            if (!entry.getName().startsWith(packagePrefix)) {
                 continue;
             }
 
-            // Copy the file over when found.
-            try (OutputStream outputStream = new FileOutputStream(targetFile)) {
-                IOUtils.copy(tarball, outputStream);
+            // Resolve path without 'package' prefix.
+            Path entryPath = Path.of(packagePrefix).relativize(Path.of(entry.getName()));
+
+            // Skip file if it's extension is not permitted.
+            String extension = getFileExtension(entryPath.getFileName().toString());
+            if(!ALLOWED_EXTENSIONS.contains(extension)) {
+                continue;
             }
+
+            // Resolve target path and copy file.
+            Path targetPath = installationPath.resolve(entryPath);
+            Files.createDirectories(targetPath.getParent());
+            Files.copy(tarball, targetPath);
+        }
+
+        // Add package to the imports so it can be written to the import map later.
+        Path importPath = Path.of("/vendor", name, latestVersion.resolveEntryPoint()).normalize();
+        imports.put(name, importPath.toString());
+    }
+
+    private void renderPage() throws Exception {
+        // Build import map from installed packages.
+        Map<String, Map<String, String>> importMap = new HashMap<>();
+        importMap.put("imports", imports);
+
+        // Ad the import map to the template attributes.
+        Map<String, Object> attributes = new HashMap<>();
+        attributes.put("importMap", JsonParser.writeValueAsString(importMap));
+
+        // Render the template.
+        File targetDir = new File(context.getTargetDir(), "app");
+        context.freeMarker().writeFile(attributes, "templates/app.ftl", targetDir, "index.html");
+    }
+
+    private String getFileExtension(String filename) {
+        int dotIndex = filename.lastIndexOf(".");
+        if (dotIndex >= 0) {
+            return filename.substring(dotIndex + 1);
         }
+        return "";
     }
 
     @Override
     protected String getTitle() {
-        return "App";
+        return "Vendor";
```

Comment:
Vendor makes no sense - this is rendering `app/index.html` so should be `App` or perhaps `Mock App` or whatever.

```suggestion
        return "App";
```

Reply:
Ah yes, forgot to rename this back, thanks.